e2e/e2etests/test_solana_deposit_call.go

```diff
@@ -0,0 +1,53 @@
+package e2etests
+
+import (
+	"math/big"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/e2e/utils"
+	testcontract "github.com/zeta-chain/zetacore/testutil/contracts"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// TestSolanaDepositAndCall tests deposit of lamports calling a example contract
+func TestSolanaDepositAndCall(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 1)
+
+	// parse deposit amount (in lamports)
+	// #nosec G115 e2e - always in range
+	depositAmount := big.NewInt(int64(parseInt(r, args[0])))
```

Comment:
Can we use setString instead? Allow to provide bigger values

https://github.com/zeta-chain/node/blob/c5868affc70b75fbb02ff80de4dc1150abd504de/e2e/e2etests/test_v2_erc20_withdraw_and_call.go#L19

Reply:
refactor in commit https://github.com/zeta-chain/node/pull/2726/commits/92d67f5897576fc119b93838893b0aaf9a43ec08